posts/ja/2023-02-13-22.0.0.13.adoc

```diff
@@ -0,0 +1,199 @@
++
+この問題は解決され、デフォルトの `HttpAuthenticationMechanisms` は認証されていないリクエストに対して `HttpMessageContext.responseUnauthorized` を適切に呼び出すようになりました。
+
+* link:https://github.com/OpenLiberty/open-liberty/issues/23146[JspFactory.getDefaultFactory().getEngineInfo().getSpecificationVersion()不正なバージョンを返す]
++
+Pages (旧 JSP) アプリケーションが `JspFactory.getDefaultFactory().getEngineInfo().getSpecificationVersion()` を呼び出すと、間違ったバージョンが返されることがあります。この値は、正しい仕様のバージョンではなく、`2.1` を返すようにコード化されています。 
++
+この問題は解決され、正しいバージョンが返されるようになりました。 例えば、`jsp-2.2` は `2.2` を返し、`pages-3.0` は `3.0` を返します。
+
+* link:https://github.com/OpenLiberty/open-liberty/issues/22405[OidcClientImplがSecurityServiceへの依存関係を正しく宣言していない]
++
+フォームベースのログアウトURLである `ibm_security_logout` へのトラフィックを遮断する `ServletFilter` から `request.logout()` を呼び出すと、以下のNPEが発生する可能性があります。:
+[source]
+----
+java.lang.NullPointerException
+Stack Dump = java.lang.NullPointerException
+        at com.ibm.ws.security.openidconnect.client.internal.OidcClientImpl.authenticateSubject(OidcClientImpl.java:749)
+        at com.ibm.ws.security.openidconnect.client.internal.OidcClientImpl.handleOidcCookie(OidcClientImpl.java:722)
+        at com.ibm.ws.security.openidconnect.client.internal.OidcClientImpl.logout(OidcClientImpl.java:663)
+        at com.ibm.ws.webcontainer.security.AuthenticateApi.logoutUnprotectedResourceServiceRef(AuthenticateApi.java:244)
+        at com.ibm.ws.webcontainer.security.AuthenticateApi.logout(AuthenticateApi.java:189)
+        at com.ibm.ws.webcontainer.security.AuthenticateApi.logoutServlet30(AuthenticateApi.java:627)
+        at com.ibm.ws.webcontainer.security.WebAppSecurityCollaboratorImpl.logout(WebAppSecurityCollaboratorImpl.java:1212)
+        at com.ibm.ws.webcontainer.srt.SRTServletRequest.logout(SRTServletRequest.java:3956)
+        at javax.servlet.http.HttpServletRequestWrapper.logout(HttpServletRequestWrapper.java:376)
+        at com.ibm.bpm.servlet.filters.GenericSecurityServletFilter.doFilter(GenericSecurityServletFilter.java:327)
+        at com.ibm.ws.webcontainer.filter.FilterInstanceWrapper.doFilter(FilterInstanceWrapper.java:201)
+----
++
+この問題は、 `OidcClientImpl` dependent on `SecurityService` を宣言することで解決され、NPE は発生しなくなりました。
+
+
+[#ガイド]
+== 前回のリリース以降の新しいガイドと更新されたガイド
+Open Libertyの特徴や機能が増え続ける中、できるだけ簡単に導入できるように、それらのトピックに関するlink:https://openliberty.io/guides/?search=new&key=tag[openliberty.ioへの新しいガイド]を追加しています。既存のガイドも、報告されたバグや問題に対処し、内容を最新に保ち、トピックの内容を拡張するために更新されます。
+
+* link:{url-prefix}/guides/jakarta-websocket.html[Jakarta WebSocketを用いたサービス間双方向通信]
+** Jakarta WebSocketを使用して、接続を終了せずにサービス間でメッセージを送受信する方法について説明します。
+* link:{url-prefix}/guides/grpc-intro.html[gRPCを用いたクライアントとサーバサービス間のメッセージのストリーミング]
+** Open Libertyを使用して、gRPC単体呼び出し、サーバストリーミング、クライアントストリーミング、双方向ストリーミングを使用して、Javaクライアントとサーバサービス間の通信を行う方法について説明します。
+
+
+== 今すぐOpen Liberty 22.0.0.13を入手する
+
+下記のリンクから入手できます <<run,Maven, Gradle, Docker, and as a downloadable archive>>.
```

Comment:
This link should look like this.  Other than that everything looks good.
![Screenshot 2023-02-20 at 9 41 45 AM](https://user-images.githubusercontent.com/84740311/220149397-5f0234ba-306b-46b5-989c-15f8af53c64b.png)

Reply:
Fixed!